dataset_template.py

```diff
@@ -0,0 +1,59 @@
+class Template:
+    """ Template object encapsulates a dictionary of template options.
+
+        Given a dataset.template file, this object creates a dictionary of options keyed on the option name. This
+        will allow for accessing of various template parameters in constant time (as opposed to O(n) time) and includes
+        a simple, user friendly syntax.
+
+        Use as follows:
+
+            template = Template(file_name)          # create a template object
+            options = template.get_options()        # access the options dictionary
+            dataset = options['dataset']            # access a specific option
+
+    """
+
+    options = {} # The options dictionary
+
+    def __init__(self, template_file):
+        """ Initializes Template object with a template file.
+
+            The provided template file is parsed line by line, and each option is added to the options dictionary.
+
+            :param template_file: file, the template file to be accessed
+        """
+
+        # Adds the dataset name as parsed from the filename to the dictionary for easy lookup
+        self.options['dataset'] = template_file.split('/')[-1].split(".")[0]
+
+        # Open files for reading
+        with open(template_file) as template:
+
```

Comment:
I guess that is a style choice. I prefer to keep logical blocks separated by new lines on either end, so that only code that is directly logically collected are blocked together.

Reply:
Just wanted to link to the PEP guidelines: https://www.python.org/dev/peps/pep-0008/#blank-lines. Food for thought for the future.